block_iagora.php

```diff
@@ -66,12 +100,13 @@ public function get_content() {
      * @return string The generated HTML content for the chat.
      */
     private function generate_chat_content($copilotendpointurl) {
-        global $OUTPUT;
+        global $OUTPUT, $USER, $COURSE;
```

Comment:
```suggestion
        global $OUTPUT;
```
Where are `$USER` and `$COURSE` used?

Reply:
Dans `get_student_progress()` j'ai utilisé `$COURSE` et `$USER` pour récupérer les informations de progression des activités de l'utilisateur `get_fast_modinfo($COURSE, $USER->id)` et aussi dans`$completion->get_data($cm, true, $USER->id)` pour récupérer si l'utilisateur a complété ou pas l'activité.